### PR TITLE
WT-9831 Add tests to new variant Amazon Linux 2

### PR DESCRIPTION
**Pull request by etienneptl**
Added a new variant `amazon2-arm64`
The new variant contains the same tests as `ubuntu2004-arm64`.
There are failing tests, tickets have been created to address them. Until then, they are disabled.

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -5334,3 +5334,43 @@ buildvariants:
     - name: unit-test-nonstandalone
     - name: unit-test-long-nonstandalone
       distros: ubuntu2004-arm64-large
+
+- name: amazon2-arm64
+  display_name: "~ Amazon Linux 2 ARM64"
```
Comment (lukech) [inline test/evergreen.yml:5339]: The leading `~` usually indicates a builder is either the least important or most expensive - neither applies to this builder?

File: test/evergreen.yml [hunk 2]
```diff
@@ -5334,3 +5334,43 @@ buildvariants:
...
+    test_env_vars:
+      WT_TOPDIR=$(git rev-parse --show-toplevel)
+      WT_BUILDDIR=$WT_TOPDIR/cmake_build
+      LD_LIBRARY_PATH=$WT_BUILDDIR
+    CMAKE_INSTALL_PREFIX: -DCMAKE_INSTALL_PREFIX=$(pwd)/cmake_build/LOCAL_INSTALL
+    python_binary: '/opt/mongodbtoolchain/v3/bin/python3'
+    smp_command: -j $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
+    cmake_generator: "Unix Makefiles"
+    make_command: make
+  tasks:
```
Comment (lukech) [inline test/evergreen.yml:5355]: Can we add in a few non-standalone tasks, to simulate the MDB cloud environment? I'd suggest starting with `make-check-nonstandalone` and `unit-test-nonstandalone`.

Comment (lukech) [reply test/evergreen.yml:5355]: Actually, if non-standalone tasks are needed, the corresponding standalone tasks can be saved. I believe the main purpose of introducing this new builder is to better match the Atlas environment.

Comment (etienneptl) [reply test/evergreen.yml:5355]: I forgot about those. I added the same fleet of tests we have for ubuntu2004-arm64, see e9da8eb476cd1fb6899625b2dd0d881bb882a1b2

Comment (lukech) [reply test/evergreen.yml:5355]: Thanks.

**Review by lukech — APPROVED** [review]
LGTM

**Review by mickgraham-mongodb — APPROVED** [review]
LGTM

**Pull request merged**